
<!DOCTYPE HTML>

<html lang="en">

<head>
  <meta charset="utf-8">
  <title>Std&#58;&#58;promise&#58;&#58;set_exception_at_thread_exit - C++ - W3cubDocs</title>
  
  <meta name="description" content="Stores the exception pointer p into the shared state without making the state ready immediately. The state is made ready when the current thread &hellip;">
  <meta name="keywords" content="std, promise, set, exception, at, thread, exit, -, c++, cpp">
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="mobile-web-app-capable" content="yes">
  
  <link rel="canonical" href="http://docs.w3cub.com/cpp/thread/promise/set_exception_at_thread_exit/">
  <link href="/favicon.png" rel="icon">
  <link type="text/css" rel="stylesheet" href="/assets/application-50364fff564ce3b6327021805f3f00e2957b441cf27f576a7dd4ff63bbc47047.css">
  <script type="text/javascript" src="/assets/application-db64bfd54ceb42be11af7995804cf4902548419ceb79d509b0b7d62c22d98e6f.js"></script>
  <script src="/json/cpp.js"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-71174418-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body>
	<div class="_app">
	<header class="_header">
  
  <form class="_search">
    <input type="search" class="_search-input" placeholder="Search&hellip;" autocomplete="off" autocapitalize="off" autocorrect="off" spellcheck="false" maxlength="20">
    <a class="_search-clear"></a>
    <div class="_search-tag"></div>
  </form>
  
  <a class="_home-link" href="/" ></a>
  <a class="_menu-link"></a>
  <h1 class="_logo">
    <a href="/" class="_nav-link" title="API Documentation Browser">W3cubDocs</a>
  </h1>
  
  <span class="_logo-sub-nav">/</span><span class="_logo-sub-nav"><a href="/cpp/" class="_nav-link" title="" style="margin-left:0;">C++</a></span>
  
  <nav class="_nav">
    <a href="/app/" class="_nav-link ">App</a>
    <a href="/about/" class="_nav-link ">About</a>
  </nav>
</header>
	<section class="_sidebar">
		<div class="_list">
			
		</div>
	</section>
	<section class="_container ">
		<div class="_content">
			<div class="_page _c">
				
<h1 id="firstHeading" class="firstHeading">std::promise::set_exception_at_thread_exit</h1>            <table class="t-dcl-begin"><tbody> <tr class="t-dcl t-since-cxx11"> <td> <pre data-language="cpp">void set_exception_at_thread_exit( std::exception_ptr p );</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx11">(since C++11)</span> </td> </tr>  </tbody></table> <p>Stores the exception pointer <code>p</code> into the shared state without making the state ready immediately. The state is made ready when the current thread exits, after all variables with thread-local storage duration have been destroyed.</p>
<p>The operation behaves as though <code><a href="../set_value/" title="cpp/thread/promise/set value">set_value</a></code>, <code><a href="../set_exception/" title="cpp/thread/promise/set exception">set_exception</a></code>, <code><a href="../set_value_at_thread_exit/" title="cpp/thread/promise/set value at thread exit">set_value_at_thread_exit</a></code>, and <code>set_exception_at_thread_exit</code> acquire a single mutex associated with the promise object while updating the promise object.</p>
<p>An exception is thrown if there is no shared state or the shared state already stores a value or exception.</p>
<h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> p </td> <td> - </td> <td> exception pointer to store. The behavior is undefined if <code>p</code> is null. </td>
</tr>
</table> <h3 id="Return_value"> Return value</h3> <p>(none).</p>
<h3 id="Exceptions"> Exceptions</h3> <p><code><a href="../../future_error/" title="cpp/thread/future error">std::future_error</a></code> on the following conditions:</p>
<ul>
<li> <code>*this</code> has no shared state. The error category is set to <a href="../../future_errc/" title="cpp/thread/future errc"><code>no_state</code></a>. </li>
<li> The shared state already stores a value or exception. The error category is set to <a href="../../future_errc/" title="cpp/thread/future errc"><code>promise_already_satisfied</code></a>. </li>
</ul>  <h3 id="Example"> Example</h3>  <h3 id="See_also"> See also</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div class="t-dsc-member-div"><div><a href="../set_exception/" title="cpp/thread/promise/set exception"> <span class="t-lines"><span>set_exception</span></span></a></div></div> </td> <td> sets the result to indicate an exception <br> <span class="t-mark">(public member function)</span>  </td>
</tr> </table>           <div class="_attribution">
  <p class="_attribution-p">
    Â© cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="http://en.cppreference.com/w/cpp/thread/promise/set_exception_at_thread_exit" class="_attribution-link" target="_blank">http://en.cppreference.com/w/cpp/thread/promise/set_exception_at_thread_exit</a>
  </p>
</div>

			</div>
		</div>
	</section>

	</div>
</body>
</html>
